test.py

```diff
@@ -1740,8 +1799,10 @@ async def main() -> int:
     print_summary(failed_tests, options)
 
     for mode in options.modes:
+        junit_file = f"{options.tmpdir}/{mode}/allure/boost.junit.xml"
         write_junit_report(options.tmpdir, mode)
-        write_consolidated_boost_junit_xml(options.tmpdir, mode)
+        xunit_file = write_consolidated_boost_junit_xml(options.tmpdir, mode)
+        boost_to_junit(xunit_file, junit_file)
```

Comment:
Q: won't jenkins see both junix and boost-style, and double-count results?

Reply:
Currently, Jenkins gets reports from testlog/arch/xml directory. Allure will be checking directories testlog/arch/allure, so none of the reports will have duplication. This is done to avoid issue with double counting the test results.

Reply:
But why generate both? If everything supports xunit, let's switch to that.

Reply:
And having two different results directories is an invitation to mismatch.

Reply:
> But why generate both? If everything supports xunit, let's switch to that.

There is additional logic done for boost tests. Method [summarize_boost_tests](https://github.com/scylladb/scylladb/blob/b72906518f44844a69ee2f682dc574cb06d4f74a/test.py#L1601) checking which test is failed, how many times and instead of returning the three failed tests, it will return one with info message that test failed three times and in which mode. This aggregation easier to do from XML format because in JUnit all this information will be in a string.
<details>
<summary>Boost XML report example</summary>

```
<TestLog>
  <TestSuite name="Master Test Suite">
    <TestCase name="test_region_groups" skipped="yes" reason="disabled"/>
    <TestCase name="test_region_groups_basic_throttling" skipped="yes" reason="disabled"/>
    <TestCase name="test_region_groups_fifo_order" skipped="yes" reason="disabled"/>
    <TestCase name="test_region_groups_basic_throttling_simple_active_reclaim" skipped="yes" reason="disabled"/>
    <TestCase name="test_region_groups_basic_throttling_active_reclaim_worst_offender" skipped="yes" reason="disabled"/>
    <TestCase name="test_no_crash_when_a_lot_of_requests_released_which_change_region_group_size" file="test/boost/dirty_memory_manager_test.cc" line="432">
      <Info file="seastar/src/testing/seastar_test.cc" line="43">check true has passed</Info>
      <Info file="test/boost/dirty_memory_manager_test.cc" line="473">check !f.available() has passed</Info>
      <Info file="test/boost/dirty_memory_manager_test.cc" line="473">check !f.available() has passed</Info>
      .......
      <Info file="test/boost/dirty_memory_manager_test.cc" line="473">check !f.available() has passed</Info>
      <Info file="test/boost/dirty_memory_manager_test.cc" line="473">check !f.available() has passed</Info>
      <TestingTime>2265227</TestingTime>
    </TestCase>
    <TestCase name="test_reclaiming_runs_as_long_as_there_is_soft_pressure" skipped="yes" reason="disabled"/>
    <TestCase name="test_size_tracked_region_move" skipped="yes" reason="disabled"/>
    <TestCase name="test_size_tracked_region_move_assign" skipped="yes" reason="disabled"/>
  </TestSuite>
</TestLog>
```
</details>


<details>
<summary>JUnit report example</summary>

```
<testsuite tests="1" skipped="8" errors="0" failures="0" id="0" name="Master_Test_Suite" time="2.26528">
    <testcase assertions="0" name="test_region_groups" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_region_groups'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_region_groups_basic_throttling" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_region_groups_basic_throttling'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_region_groups_fifo_order" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_region_groups_fifo_order'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_region_groups_basic_throttling_simple_active_reclaim" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_region_groups_basic_throttling_simple_active_reclaim'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_region_groups_basic_throttling_active_reclaim_worst_offender" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_region_groups_basic_throttling_active_reclaim_worst_offender'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="1000001" name="test_no_crash_when_a_lot_of_requests_released_which_change_region_group_size" time="2.26523">
        <system-out><![CDATA[INFO:
- file   : seastar_test.cc
- line   : 43
- message: check true has passed

INFO:
- file   : dirty_memory_manager_test.cc
- line   : 473
- message: check !f.available() has passed

......

INFO:
- file   : dirty_memory_manager_test.cc
- line   : 473
- message: check !f.available() has passed

INFO:
- file   : dirty_memory_manager_test.cc
- line   : 473
- message: check !f.available() has passed

]]></system-out>
    </testcase>
    <testcase assertions="0" name="test_reclaiming_runs_as_long_as_there_is_soft_pressure" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_reclaiming_runs_as_long_as_there_is_soft_pressure'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_size_tracked_region_move" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_size_tracked_region_move'
- reason: 'disabled']]></system-out>
    </testcase>
    <testcase assertions="0" name="test_size_tracked_region_move_assign" time="0">
        <skipped/>
        <system-out><![CDATA[Test case disabled because of the following chain of decision:
- disabled test unit: 'test_size_tracked_region_move_assign'
- reason: 'disabled']]></system-out>
    </testcase>
</testsuite>
```
</details> 


> And having two different results directories is an invitation to mismatch.

Unfortunately, we will need to have two directories. Allure can use information from its own format and from JUnit. The Jenkins plugin can consume JUnit/NUnit. The problem with one directory, that allure will consume its own format and JUnit output, so we will end with duplicates in the final report. I can see here several solutions (taking in account we already separated python JUnit output and allure output):
1. Leave XML format for boost, and convert the generated output from test.py to JUnit format. **<--- Current solution**
    - Pros:
      - Identical reports since we are converting the existing one
    - Cons:
      - We are still using the conversion between XML and NUnit, but this functionality already there
2. Add JUnit logger that will output to allure directory.
    - Pros:
      - Supported format for allure and Jenkins plugin
    - Cons:
      - Cleaning of the skipped test cases needed. We are executing boost test by function, so we will generate a lot of skipped tests in the report. Example: file A contains 3 test cases(a,b,c). So there will be 3 reports: file_A_test_a report with 'a' passed and 'b','c' skipped, file_A_test_b report with 'b' passed and 'a','c' skipped, file_A_test_c report with 'c' passed and 'a','b' skipped. So we need to clean this not really skipped cases. XML output already have this.
    - If we want to have the same format, some summarizing needed.
3. Switch off XML and switch on JUnit.
    - Pros:
       - Fewer conversions between different formats
       - Jenkins NUnit plugin can be uninstalled 
    - Cons:
        - Method for summarization should be modified
        - JUnit will be less convenient to modify than XML